ultravox/data/types.py

```diff
@@ -0,0 +1,206 @@
+    splits: Optional[List[DatasetSplitConfig]] = None
+    """List of splits to use, e.g. [{"name": "train", "num_samples": 1000}, {"name": "validation", "num_samples": 100}]."""
+    user_template: Optional[str] = None
+    """Template for the user message."""
+    user_template_args: Optional[Dict[str, str]] = None
+    """Optional arguments (e.g., target language) for the user template."""
+    assistant_template: Optional[str] = None
+    """Template for the assistant message."""
+    transcript_template: Optional[str] = None
+    """Template for the transcript."""
+    audio_field: Optional[str] = None
+    """Field in the dataset that contains the audio, use None if the dataset does not contain audio."""
+    use_mds: Optional[bool] = None
+    """Set to True to load the dataset from GCP (using MDS) instead of Hugging Face."""
+    mds_batch_size: Optional[int] = None
+    """Batch size for the dataset when using MDS."""
+
+    def __post_init__(self):
+        """Set defaults only if this is a root config, so that said defaults in a subclass don't act as overrides."""
+        DEFAULTS = {
+            "splits": [],
+            "user_template": AUDIO_PLACEHOLDER,
+            "user_template_args": {},
+            "assistant_template": "{{text}}",
+            "transcript_template": "{{text}}",
+            "audio_field": "audio",
+            "use_mds": False,
+            "mds_batch_size": 32,
+        }
+        if self.base is None:
+            for attr, default_value in DEFAULTS.items():
+                if getattr(self, attr) is None:
+                    setattr(self, attr, default_value)
+
+
+def audio_from_file(path: str) -> np.ndarray:
+    """Load audio from a file, converting to float32 PCM @ 16 kHz."""
+    audio, _ = librosa.load(path, sr=SAMPLE_RATE)
+    assert audio.dtype == np.float32
+    return audio
+
+
+def audio_from_buf(buf: bytes) -> np.ndarray:
+    """Load audio from a buffer, converting to float32 PCM @ 16 kHz."""
+    audio, _ = librosa.load(io.BytesIO(buf), sr=SAMPLE_RATE)
+    assert audio.dtype == np.float32
+    return audio
+
+
+def audio_to_wav(audio: np.ndarray, sample_rate: int = SAMPLE_RATE) -> bytes:
+    """Convert audio to WAV format, 16-bit PCM @ 16 kHz."""
+    assert audio.dtype == np.float32
+    with io.BytesIO() as buf:
+        sf.write(buf, audio, sample_rate, format="WAV", subtype="PCM_16")
+        return buf.getvalue()
+
+
+def audio_to_wav_base64(audio: np.ndarray, sample_rate: int = SAMPLE_RATE) -> str:
+    """Convert audio to a base64-encoded WAV file."""
+    return base64.b64encode(audio_to_wav(audio, sample_rate)).decode("utf-8")
+
+
+def audio_to_data_uri(audio: np.ndarray, sample_rate: int = SAMPLE_RATE) -> str:
+    """Convert audio to a data URI."""
+    return f"data:audio/wav;base64,{audio_to_wav_base64(audio, sample_rate)}"
+
+
+def messages_from_prompt(prompt: str) -> List[Dict[str, str]]:
+    return [{"role": "user", "content": prompt}]
```

Comment:
Should we define these methods in a separate file, like `helpers.py`, as they are not data `types`.

Reply:
yeah, fair point, I can move this to a new file like `sample.py`.